lib/active_model/serializer.rb

```diff
@@ -191,10 +191,24 @@ def json_key
     end
 
     def read_attribute_for_serialization(attr)
+      # Start by favoring serializers method override
       if respond_to?(attr)
         send(attr)
       else
-        object.read_attribute_for_serialization(attr)
+        # Otherwise, check read_attribute_for_serialization
+        if object.respond_to?(:read_attribute_for_serialization)
+          object.read_attribute_for_serialization(attr)
```

Comment:
The `object` must *always* respond to `read_attribute_for_serialization`. That's part of the contract.  I'm not sure why we'd want to weaken that.  I'd rather provide opt-in mixins for `Hash` for those that are trying to serialize a `primitive` (which you shouldn't be doing).